spaceship/lib/spaceship/tunes/availability.rb

```diff
@@ -51,16 +62,42 @@ def self.from_territories(territories = [], params = {})
         obj.include_future_territories = params.fetch(:include_future_territories, true)
         obj.cleared_for_preorder = params.fetch(:cleared_for_preorder, false)
         obj.app_available_date = params.fetch(:app_available_date, nil)
+        obj.b2b_unavailable =  params.fetch(:b2b_unavailable, false)
+        obj.b2b_app_enabled =  params.fetch(:b2b_app_enabled, false)
         return obj
       end
 
       def territories
         @territories ||= raw_data['countries'].map { |info| Territory.new(info) }
       end
 
+      def b2b_users
+        @b2b_users ||= raw_data['b2bUsers'].map { |user| B2bUser.new(user) }
+      end
+
       def cleared_for_preorder
         super || false
       end
+
+      # Sets the b2b flag. If you call Save on app_details without adding any b2b users
+      # it will result in an error.
+      def enable_b2b_app!
+        print("b2b : " + b2b_unavailable.to_s)
+        raise "Not possible to enable b2b on this app" if b2b_unavailable
+        @b2b_app_enabled = true
+        return self
+      end
+
+      # just adds to the availability, You will still have to call update_availabilty
+      def add_b2b_users(user_list = [])
+        raise "Cannot add b2b users if b2b is not enabled" unless b2b_app_enabled
+        b2b_user_array = []
+        user_list.each do |user|
+          b2b_user_array.push(B2bUser.from_username(user))
+        end
+        @b2b_users = b2b_user_array
+        return self
```

Comment:
This method looks like its "setting" users instead of "adding" users. Should this method be renamed? Also, this method could be cleaned up to look like the follow if it is actually just "setting"...

```rb
def add_b2b_users(user_list = [])
  raise "Cannot add b2b users if b2b is not enabled" unless b2b_app_enabled
  @b2b_users = map do |user|
    B2bUser.from_username(user)
  end
  return self
end
```

Reply:
@joshdholtz its peculiar but when i examined the request body, only the ones that were being added or removed were included in the request body. so this is indeed "adding" instead of "setting". But your comment about "cleanup" stands. I will make the change.